python/lsst/ts/athexapod/controller.py

```diff
@@ -80,38 +72,32 @@ def is_connected(self):
         connected : `bool`
 
         """
-        return self.reader is not None and self.writer is not None
+        return self.client.connected
 
     async def connect(self):
         """Connect to hexapod controller."""
 
-        async with self.lock:
-            if self.is_connected:
-                raise RuntimeError(
-                    "Reader or Writer not None. Try disconnecting first."
-                )
-
-            connect_task = asyncio.open_connection(host=self.host, port=self.port)
-
-            self.reader, self.writer = await asyncio.wait_for(
-                connect_task, timeout=self.long_timeout
-            )
+        self.client = tcpip.Client(
+            host=self.host,
+            port=self.port,
+            log=self.log,
+            name="ATHexapod client",
+            encoding="ISO-8859-1",
+            terminator=b"\n",
+        )
+        await self.client.start_task
 
     async def disconnect(self):
         """Disconnect from hexapod controller."""
 
-        async with self.lock:
-            self.reader = None
-
-            if self.writer is not None:
-                try:
-                    self.writer.write_eof()
-                    await asyncio.wait_for(self.writer.drain(), timeout=self.timeout)
-                finally:
-                    self.writer.close()
-                    self.writer = None
+        try:
+            await self.client.close()
+        except Exception:
+            self.log.exception("Failed to disconnect")
+        finally:
+            self.client = tcpip.Client(host="", port=None, log=self.log)
```

Comment:
Uhm.. why are you recreating the client here? shouldn't this be set to `None`?

Reply:
The way that the control flow works is that the client is initialized with an empty host and then recreate with the correct configuration and then recreated to empty host because I want to know whether the client is connected at any time as opposed to using None.